sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -78,6 +78,9 @@ class SQLContext private[sql](
   }
   def this(sparkContext: JavaSparkContext) = this(sparkContext.sc)
 
+  // Make sure the sparkContext we have at here is not stopped.
+  sparkContext.assertNotStopped
```

Comment:
@pwendell @rxin @marmbrus 

@mengxr and I had a quick discussion. We think it is generally good to make sure the `sparkContext` provided to a `SQLContext` is not stopped. To do so, we'd like change the access modifier of `assertNotStopped` to `private[spark]`. Is this change good?